spec/models/spree/payment/processing_decorator_spec.rb

```diff
@@ -14,5 +14,12 @@
 
       expect(payment.close!).to be_truthy
     end
+
+    it 'payment receives close! method' do
+      payment = create(:amazon_payment, state: 'completed')
+
+      expect(payment).to receive(:close!)
```

Comment:
When I do this it passes:

`
it 'delegates close! to the payment source' do
      payment = create(:amazon_payment, state: 'completed')
      allow_any_instance_of(Spree::AmazonTransaction).to receive(:close!)

      payment.close!

      expect(payment.source).to have_received(:close!)
    end
`

Any suggestions on a different approach?